### PR TITLE
Remove duplication in configvalidator #578

### PR DESCRIPTION
**Pull request by rahulgoyal030**
Fixes: issue #578 
Licence: Released under MIT License (MIT)
description : created a single method "isSortingSafe()" which can now check both 'type' and 'order' of the sorting.

### CODE REVIEW

**Comment by oparoz** [comment]
I'm disappointed as it was easier than I initially thought :D 

Good job on the PR description.
You should pick the MIT license for such small changes until you register as an ownCloud contributor, then you can release your work under the AGPL if you want to.

I'll comment inline.

File: config/configvalidator.php [hunk 1]
```diff
@@ -32,8 +32,8 @@ public function isConfigSafe($key, $parsedConfigItem) {
 
 		switch ($key) {
 			case 'sorting':
-				$safe = $this->isSortingTypeSafe($parsedConfigItem, $safe);
-				$safe = $this->isSortingOrderSafe($parsedConfigItem, $safe);
+				$safe = $this->isSortingSafe( 'type',$parsedConfigItem, $safe);
```
Comment (oparoz) [inline config/configvalidator.php]: spacing.
Make sure to ask your IDE to format the text before making the commit.

File: config/configvalidator.php [hunk 2]
```diff
@@ -48,32 +48,18 @@ public function isConfigSafe($key, $parsedConfigItem) {
 	 *
 	 * @param array $parsedConfigItem the sorting configuration to analyse
 	 * @param bool $safe whether the current config has been deemed safe to use so far
-	 *
+	 * @param string key  whether this exist or not
```
Comment (oparoz) [inline config/configvalidator.php]: typo and could be more explicit. Think about the other developers who will read your code, they need to be able to quickly understand what is going on by reading the documentation.

File: config/configvalidator.php [hunk 3]
```diff
@@ -48,32 +48,18 @@ public function isConfigSafe($key, $parsedConfigItem) {
...
-	 * @return bool
-	 */
-	private function isSortingOrderSafe($parsedConfigItem, $safe) {
-		if ($safe && array_key_exists('order', $parsedConfigItem)) {
-			$safe = $safe && $this->sortingValidator('order', $parsedConfigItem['order']);
-		}
-
-		return $safe;
-	}
+
```
Comment (oparoz) [inline config/configvalidator.php]: Too much white space. Not a big deal, of course, but maybe your IDE can be configured to avoid it?

Comment (rahulgoyal030) [inline config/configvalidator.php]: sorry , i will  definitely do better next time . learn a lot from this and thanks for your valuable suggestions :)

Comment (oparoz) [inline config/configvalidator.php]: No worries. I know it's a bit frustrating at the beginning and everybody's got different preferences, but I find that the cleaner we can keep the doc and the comments, the better it is for the project :)

**Comment by oparoz** [comment]
@rahulgoyal030 - Please ping testers when you've made changes and are expecting reviews, otherwise we miss it ;)

**Comment by rahulgoyal030** [comment]
@oparoz  is this  okay or  it can be further improved ?

**Comment by oparoz** [comment]
No, it's OK, I just need to test it one more time and you need to figure out under which license you want to release this.

**Comment by oparoz** [comment]
There has been a slight misunderstanding, you don't need to ship a licence file, just say that this is release under the MIT and it's fine.

**Comment by oparoz** [comment]
@rahulgoyal030 - Thanks. For next time, please make sure your IDE adds a space after a comma in method signatures :)

**Comment by rahulgoyal030** [comment]
@oparoz  - yeah  ,  i will :)

**Pull request merged**